sky/task.py

```diff
@@ -871,8 +869,8 @@ def get_preferred_store_type(self) -> storage_lib.StoreType:
         # assert len(self.resources) == 1, self.resources
         storage_cloud = None
 
-        backend_utils.check_public_cloud_enabled()
-        enabled_storage_clouds = global_user_state.get_enabled_storage_clouds()
+        enabled_storage_clouds = storage_lib.get_enabled_storage_clouds(
+            raise_if_no_cloud_access=True)
         if not enabled_storage_clouds:
             raise ValueError('No enabled cloud for storage, run: sky check')
```

Comment:
It seems the `raise_if_no_cloud_access=True` is duplicated with the `if not enabled_storage_clouds` below.